crates/sui-storage/src/node_sync_store.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::default_db_options;
+use std::path::Path;
+
+use sui_types::{
+    base_types::TransactionDigest,
+    error::{SuiError, SuiResult},
+    messages::{CertifiedTransaction, SignedTransactionEffects},
+};
+
+use typed_store::rocks::DBMap;
+use typed_store::{reopen, traits::Map};
+
+/// NodeSyncStore store is used by nodes to store downloaded objects (certs, etc) that have
+/// not yet been applied to the node's SuiDataStore.
+pub struct NodeSyncStore {
+    /// Certificates/Effects that have been fetched from remote validators, but not sequenced.
+    certs_and_fx: DBMap<TransactionDigest, (CertifiedTransaction, SignedTransactionEffects)>,
+}
```

Comment:
Do we want this to actually be a persisted database or just an in memory one? I only ask because the data we write to rocksdb keeps increasing :)

Reply:
I think its probably worth it to save the extra downloads in the case where the node is being restarted or crashing. This store is garbage collected very quickly (as soon as we apply the cert locally we delete the entry here), so it shouldn't consume much space.